python/taichi/lang/ast/ast_transformer.py

```diff
@@ -468,9 +469,22 @@ def build_Return(ctx, node):
                         f'A {"kernel" if ctx.is_kernel else "function"} '
                         'with a return value must be annotated '
                         'with a return type, e.g. def func() -> ti.f32')
-                _ti_core.create_kernel_return(
-                    ti_ops.cast(expr.Expr(node.value.ptr),
-                                ctx.func.return_type).ptr)
+                if id(ctx.func.return_type) in primitive_types.type_ids:
+                    _ti_core.create_kernel_exprgroup_return(
+                        expr.make_expr_group(
+                            ti_ops.cast(expr.Expr(node.value.ptr),
+                                        ctx.func.return_type).ptr))
+                else:
+                    _ti_core.create_kernel_exprgroup_return(
+                        expr.make_expr_group([
+                            ti_ops.cast(
+                                exp, ctx.func.return_type.dtype if isinstance(
+                                    ctx.func.return_type, MatrixType) else
+                                ctx.func.return_type) for exp in list(
```

Comment:
1. What does the `else` case mean?
2. `list(` is unnecessary.

Reply:
It seems unnecessary in this PR, I will delete it.